tensorboard/plugins/scalar/scalars_demo.py

```diff
@@ -133,27 +114,46 @@ def run_all(logdir, verbose=False):
       verbose: if true, print out each run's name as it begins
     """
     for initial_temperature in [270.0, 310.0, 350.0]:
-        for final_temperature in [270.0, 310.0, 350.0]:
+        for ambient_temperature in [270.0, 310.0, 350.0]:
             for heat_coefficient in [0.001, 0.005]:
                 run_name = "t0=%g,tA=%g,kH=%g" % (
                     initial_temperature,
-                    final_temperature,
+                    ambient_temperature,
                     heat_coefficient,
                 )
                 if verbose:
                     print("--- Running: %s" % run_name)
-                run(
-                    logdir,
-                    run_name,
-                    initial_temperature,
-                    final_temperature,
-                    heat_coefficient,
+                writer = tf.summary.create_file_writer(
+                    os.path.join(logdir, run_name)
                 )
+                with writer.as_default():
+                    run(
+                        tf.constant(initial_temperature),
```

Comment:
Maybe worth a comment about why the `tf.constant()`s are here? I.e. to prevent unwanted retracing. We could also just specify `input_signature` on the `@tf.function` decorator which might be a little more foolproof.